Demo/src/main/java/com/braintreepayments/demo/VenmoFragment.java

```diff
@@ -35,14 +37,20 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
         venmoButton = view.findViewById(R.id.venmo_button);
         venmoButton.setOnClickListener(this::launchVenmo);
 
-        braintreeClient = getBraintreeClient();
-        venmoClient = new VenmoClient(this, braintreeClient);
-        venmoClient.setListener(this);
+        venmoLauncher = new VenmoLauncher(this, venmoAuthChallengeResult ->
+                venmoClient.tokenizeVenmoAccount(venmoAuthChallengeResult, this::handleVenmoResult));
 
         return view;
     }
 
-    private void handleVenmoResult(VenmoAccountNonce venmoAccountNonce) {
+    private void handleVenmoResult(VenmoAccountNonce nonce, Exception error) {
+        if (nonce != null) {
+            handleVenmoAccountNonce(nonce);
+        } else {
+            handleError(error);
+        }
```

Comment:
This isn't relevant to this PR, but worth mentioning. I remember from that one discussion a merchant dev mentioned nullability in callbacks being a hinderance. It may be worth considering a single result type for result callbacks so that `null` checks aren't required.

I recently thought about having an interface type in PPCP e.g. `VenmoResult` that would allow logic like:
```java
if (result instanceof VenmoSuccessResult) {
  // handle success
} else if (result instanceof VenmoErrorResult) {
  // handle error
}
``` 
There are other ways too, like using enums. Just a thought I wanted to share. We have to figure something out on PPCP too. It'll be a good topic to brainstorm on going forward.

Reply:
Ah yes I remember that feedback - I think a single result type could be a good move. I'll go forward with these API changes with the dual results for now and create a ticket to reconsider the approach and align across all modules